### ISSUE TITLE
RemarkCreatorPlugin fails silently in onSubmit

### ISSUE THREAD

**Issue by nibtime** [tinacms/tinacms#619]
## Description
I tried out the RemarkCreatorPlugin and followed the steps from [this blog post](https://tinacms.org/blog/add-and-delete-files/). I have substitute pages for missing languages and thought they are the perfect context to add a plugin that provides a translation for the missing language 

### Steps to reproduce
I really don't know. It will probably work fine in demos and starters. The file path does exists and is relative to repo root. I tried out other paths too, none worked. 
 I copied the class into my project and stepped through it with Chrome debugger and it definitely happens in onChange, which is weird since it is called all time when editing.
(https://github.com/tinacms/tinacms/blob/2a45c41fdba50246f04b491587ed0b1f713833cf/packages/gatsby-tinacms-remark/src/remark-creator-plugin.ts#L85)

Code I wrote looks like this. Maybe there is some gotcha I didn't realize?
```typescript
const { currentPageId, currentLocaleCode, languageData } = useLocalization()

  const createTranslationPlugin = new RemarkCreatorPlugin({
    label: "Translate Page",
    filename: () => `frontend/content/pages/${currentPageId}.${currentLocaleCode}.md`,
    body: fields => fields.content,
    frontmatter: fields => ({
      title: fields.title,
      description: fields.description,
      localeCode: currentLocaleCode,
      pageID: currentPageId
    }),
    fields: [
      {
        name: "title",
        label: "Title",
        description: `Title of the page in ${languageData.nameLocalized}`,
        // @ts-ignore
        required: true,
        component: "text"
      },
      {
        name: "description",
        label: "Description",
        component: "textarea",
        description: `Description for the page in ${languageData.nameLocalized}`
      },
      {
        name: "content",
        label: "Markdown Content",
        component: "markdown",
        description: `Actual content of the page in ${languageData.nameLocalized}`
      }
    ]
  })

  usePlugin(createTranslationPlugin)
```

### Expected result
A new file with the data entered in the create form should be created at the specified path

### Actual result
No file is created, `onSubmit` call fails silently. After clicking the create button, Tina stops working. Changes in the side bar are no longer written to the file system and the development server has to be restarted that Tina works again.

![creator_plugin_fail](https://user-images.githubusercontent.com/52962482/72567913-31915480-38b7-11ea-8b95-36a45ac3c46d.gif)

#### Tinacms Package Versions
```
├─ @tinacms/api-git@0.5.0
├─ @tinacms/core@0.5.1
├─ @tinacms/fields@0.1.17
│  ├─ @tinacms/core@^0.5.1
│  ├─ @tinacms/icons@^0.3.9
│  ├─ @tinacms/styles@^0.1.0
├─ @tinacms/form-builder@0.2.9
│  ├─ @tinacms/core@^0.5.1
│  ├─ @tinacms/react-core@^0.2.1
├─ @tinacms/git-client@0.4.1
├─ @tinacms/icons@0.3.9
├─ @tinacms/react-core@0.2.1
│  ├─ @tinacms/core@^0.5.1
├─ @tinacms/styles@0.1.0
│  ├─ gatsby-plugin-tinacms@^0.1.17
│  ├─ gatsby-tinacms-git@^0.2.15
│  ├─ gatsby-tinacms-json@^0.8.1
│  ├─ gatsby-tinacms-remark@^0.6.1
├─ gatsby-plugin-tinacms@0.1.17
│  ├─ @tinacms/core@^0.5.1
│  └─ tinacms@^0.11.0
├─ gatsby-tinacms-git@0.2.15
│  ├─ @tinacms/api-git@^0.5.0
│  ├─ @tinacms/core@^0.5.1
│  ├─ @tinacms/git-client@^0.4.1
│  ├─ gatsby-plugin-tinacms@^0.1.17
│  └─ tinacms@^0.11.0
├─ gatsby-tinacms-json@0.8.1
│  ├─ @tinacms/core@^0.5.1
│  ├─ gatsby-plugin-tinacms@^0.1.17
│  └─ tinacms@^0.11.0
├─ gatsby-tinacms-remark@0.6.1
│  ├─ @tinacms/core@^0.5.1
│  ├─ @tinacms/form-builder@^0.2.9
│  └─ tinacms@^0.11.0
├─ tinacms@0.11.0
│  ├─ @tinacms/core@^0.5.1
│  ├─ @tinacms/fields@^0.1.17
│  ├─ @tinacms/form-builder@^0.2.9
│  ├─ @tinacms/icons@^0.3.9
│  ├─ @tinacms/react-core@^0.2.1
│  ├─ @tinacms/styles@^0.1.0
```
#### Gatsby Package Versions
```
├─ @gatsbyjs/relay-compiler@2.0.0-printer-fix.4
├─ babel-preset-gatsby@0.2.11
│  ├─ gatsby-image@^2.0.23
│  ├─ gatsby-plugin-env-variables@^1.0.1
│  ├─ gatsby-plugin-purgecss@^4.0.0
│  ├─ gatsby-plugin-react-helmet@^3.0.4
│  ├─ gatsby-plugin-sass@^2.0.7
│  ├─ gatsby-plugin-sharp@^2.2.9
│  ├─ gatsby-plugin-tinacms@^0.1.17
│  ├─ gatsby-plugin-tslint@^0.0.2
│  ├─ gatsby-plugin-typescript@^2.1.5
│  ├─ gatsby-remark-copy-linked-files@^2.0.7
│  ├─ gatsby-remark-images@^3.1.6
│  ├─ gatsby-remark-relative-images@^0.2.1
│  ├─ gatsby-source-filesystem@^2.0.26
│  ├─ gatsby-source-stripe@^3.0.4
│  ├─ gatsby-tinacms-git@^0.2.15
│  ├─ gatsby-tinacms-json@^0.8.1
│  ├─ gatsby-tinacms-remark@^0.6.1
│  ├─ gatsby-transformer-json@^2.2.25
│  ├─ gatsby-transformer-remark@^2.6.9
│  ├─ gatsby-transformer-sharp@^2.1.9
│  ├─ gatsby@^2.13.31
├─ gatsby-cli@2.7.40
│  ├─ gatsby-telemetry@^1.1.19
├─ gatsby-core-utils@1.0.6
├─ gatsby-graphiql-explorer@0.2.8
├─ gatsby-image@2.2.14
├─ gatsby-link@2.2.9
├─ gatsby-page-utils@0.0.12
├─ gatsby-plugin-env-variables@1.0.1
├─ gatsby-plugin-page-creator@2.1.12
│  ├─ gatsby-page-utils@^0.0.12
├─ gatsby-plugin-purgecss@4.0.0
├─ gatsby-plugin-react-helmet@3.1.5
├─ gatsby-plugin-sass@2.1.12
├─ gatsby-plugin-sharp@2.2.18
│  ├─ gatsby-core-utils@^1.0.6
├─ gatsby-plugin-tinacms@0.1.17
├─ gatsby-plugin-tslint@0.0.2
├─ gatsby-plugin-typescript@2.1.5
├─ gatsby-react-router-scroll@2.1.6
├─ gatsby-remark-copy-linked-files@2.1.13
├─ gatsby-remark-images@3.1.19
├─ gatsby-remark-relative-images@0.2.3
├─ gatsby-source-filesystem@2.1.18
│  ├─ gatsby-core-utils@^1.0.6
├─ gatsby-source-stripe@3.0.4
│  ├─ gatsby-core-utils@1.0.15
│  ├─ gatsby-source-filesystem@2.1.22
│  ├─ gatsby-source-filesystem@2.1.22
│  │  ├─ gatsby-core-utils@^1.0.8
├─ gatsby-telemetry@1.1.19
├─ gatsby-tinacms-git@0.2.15
│  ├─ gatsby-plugin-tinacms@^0.1.17
├─ gatsby-tinacms-json@0.8.1
│  ├─ gatsby-plugin-tinacms@^0.1.17
├─ gatsby-tinacms-remark@0.6.1
├─ gatsby-transformer-json@2.2.25
├─ gatsby-transformer-remark@2.6.19
│  ├─ gatsby-core-utils@^1.0.6
├─ gatsby-transformer-sharp@2.2.12
├─ gatsby@2.14.0
│  ├─ @gatsbyjs/relay-compiler@2.0.0-printer-fix.4
│  ├─ babel-preset-gatsby@^0.2.11
│  ├─ gatsby-cli@^2.7.40
│  ├─ gatsby-core-utils@^1.0.6
│  ├─ gatsby-graphiql-explorer@^0.2.8
│  ├─ gatsby-link@^2.2.9
│  ├─ gatsby-plugin-page-creator@^2.1.12
│  ├─ gatsby-react-router-scroll@^2.1.6
│  ├─ gatsby-telemetry@^1.1.19
```

#### Tools

| Tool    | Version |
| ------- | ------- |
| browser |  Chrome Version 79.0.3945.117 (Official Build) (64-bit)   |
| os      |   see Dockerfile below    |
| node    |   v8.16.1      |
| vscode     |   1.41.0      |
| vscode remote-containers     |  0.94.0     |

#### Docker environment
```Dockerfile
ARG NODE_VERSION=8
FROM node:${NODE_VERSION}-alpine as CI

RUN apk update && apk upgrade \
  # required platform dependencies
  # https://github.com/aripalo/gatsby-docker/blob/master/Dockerfile
  # https://github.com/lovell/sharp/blob/master/docs/install.md#alpine-linux
  # https://github.com/GoogleCloudPlatform/cloud-sdk-docker/blob/master/alpine/Dockerfile
  && apk add bash git curl openssh gnupg python py-crcmod util-linux libc6-compat \
  && apk add --update-cache \
  --repository https://alpine.global.ssl.fastly.net/alpine/edge/community/ \
  vips-dev fftw-dev build-base git-lfs \
  && rm -fr /var/cache/apk/* 

# Install Google Cloud SDK
ARG CLOUD_SDK_VERSION=258.0.0
ENV CLOUD_SDK_VERSION=$CLOUD_SDK_VERSION
ENV PATH /google-cloud-sdk/bin:$PATH

RUN curl -O https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-${CLOUD_SDK_VERSION}-linux-x86_64.tar.gz \
    && tar xzf google-cloud-sdk-${CLOUD_SDK_VERSION}-linux-x86_64.tar.gz \
    && rm google-cloud-sdk-${CLOUD_SDK_VERSION}-linux-x86_64.tar.gz \
    && gcloud config set core/disable_usage_reporting true \
    && gcloud config set component_manager/disable_update_check true \
    && gcloud config set metrics/environment github_docker_image \
    && gcloud --version

# Install Pulumi
ENV PATH /root/.pulumi/bin:$PATH
RUN curl -fsSL https://get.pulumi.com | bash \
  && pulumi version

# Install Netlify and Firebase CLI
ARG NETLIFY_CLI_VERSION=2.11.31
ENV NETLIFY_CLI_VERSION=$NETLIFY_CLI_VERSION
ARG FIREBASE_CLI_VERSION=7.11.0
ENV FIREBASE_CLI_VERSION=$FIREBASE_CLI_VERSION
RUN yarn global add \ 
  netlify-cli@${NETLIFY_CLI_VERSION} \
  firebase-tools@${FIREBASE_CLI_VERSION} \
  && yarn cache clean \
  && netlify -v && firebase -V

CMD ["bash"]
```

**Comment by nibtime** [comment]
I tested this in Gatsby Cloud and it still doesn't work. However, editing keeps working there

**Comment by ncphillips** [comment]
Hey @nibtime just got around to checking this out now. I'm not able to reproduce this. Could you open the Chrome network tab and lookup the request to create the file? I'm wondering if the request is failing. If it's not, I'm wondering what's in the response body.

Is this site public, or could you add me to the site? It would be much easier for me to debug in your specific setup. 

Also, from the Gif this looks like a pretty interesting setup!

**Comment by nibtime** [comment]
Hey @ncphillips 
sorry for not replying to this, I've been really busy setting my site up for production.
While doing that, I upgraded to node `10.18` and gatsby `2.19.12`, since support for node 8 expired.
I've upgraded Gatsby before without upgrading node, but that did not fix this problem. However, the upgrade of node seems to do the trick, the problem disappeared and files are created as they should by the creator plugin. 

Also thanks for taking interest in my project. At the moment it is not public and the repo does not just contain the Gatsby site, but basically the whole infrastructure (Firebase, payment services) as code. So naturally, tons of environment variables and multiple environments are involved. It is shareable, code contains no secrets of course, but I'd need to make it more accessible first, so develop of Gatsby works even though the infrastructure is missing. When I have some spare time I'll do that, so I can invite you to the project in upcoming issues. Thank you for the great support with Tina!
